src/Control/Abstract/Modules.hs

```diff
@@ -116,8 +116,8 @@ instance ( Has (Reader (ModuleTable (Module (ModuleResult address value)))) sig
     case op of
       Load    name  k -> askModuleTable >>= maybeM (throwLoadError (ModuleNotFoundError name)) . fmap moduleBody . ModuleTable.lookup name >>= k
       Lookup  path  k -> askModuleTable >>= k . fmap moduleBody . ModuleTable.lookup path
-      Resolve names k -> k (find (`Set.member` paths) names)
-      List    dir   k -> k (filter ((dir ==) . takeDirectory) (toList paths))
+      Resolve names k -> k (find (`Set.member` paths) (map Path.absRel names))
```

Comment:
We shouldn't be calling `absRel` on `FilePath` values that aren't literals. Can you add another case to the `LoadError` type, called something like `MalformedPathError`, that indicates failure here, and call `throwLoadError` if it's the case?

Reply:
Yes, we can, and I think we shall add LoadError type.  But it shall modify the type of `data Modules address value m k` in this PR.  Shall we modify it in this PR?  I plan to modify the definition of `Modules` in another PR because it is a bit too big for this PR.

Perhaps, in this PR, we can do
```
k (find (`Set.member` $ fmap Path.toString paths) (map names))
``` 
here to avoid some nonsence filename.

Reply:
Another PR is fine, but please file an issue about it so that we don't lose track of work we need to do.

Reply:
I did not find a way to refactor this line without getting too complex.  (As the output is Oath.absRel, it is wordy to compare with String and then convert to pathtype)  So I decide to refactor it in next PR.